doc/LibrarySplit/HowToSplit.md

```diff
@@ -0,0 +1,215 @@
+versions.
+
+Foo will also be able to release independently of Bar as either beta or GA for
+future versions.  Whenever Foo wants to ship a new version it should bump its
+dependency on Bar according to which support model it is shipping under.  If Foo
+is shipping as beta it should ship with a dependency on the latest version of Bar
+regardless of support model.  If Foo is shipping as GA it should ship with a
+dependency on the latest GA version of Bar.
+
+## Types Being Moved
+
+Each type that is moved **MUST** have an identical public API surface in the new
+package.  This includes any hidden EBN members that were put in place for
+backwards compatibility in the original package.
+
+## TypeForwardedTo
+
+To maintain binary compatibility, Foo v1.1.0 will need to use `TypeForwardedTo`.
+For organization we will create a new file called `TypesForwardedTo[NewPackage].cs`
+for each new package that Foo is splitting into.
+
+In our example a new file called `TypesForwardedToBar.cs` should be added to
+Foo v1.1.0 and this will contain a list of all types that have been moved to Bar
+v1.0.0.  This is the only package we are splitting into so its the only file we
+need to add.
+
+`TypesForwardedToBar.cs` (in Foo v1.1.0):
+```csharp
+using System.Runtime.CompilerServices;
+
+[assembly: TypeForwardedTo(typeof(Resource1))]
+[assembly: TypeForwardedTo(typeof(ResourceData1))]
+[assembly: TypeForwardedTo(typeof(Resource1Collection))]
+// Add additional forwarded dependent property-only types if they moved
+```
+
+## TypeForwardedFrom
+
+For some tooling and serializers it is helpful to know where a type was
+originally defined.  To help with this we will add the `TypeForwardedFrom`
+attribute on each type that is being moved to a new package.
+
+`Resource1.cs` (in Bar v1.0.0):
+```csharp
+using System.Runtime.CompilerServices;
+
+namespace Bar
+{
+    [TypeForwardedFrom("Foo, Version=1.0.0.0, Culture=neutral, PublicKeyToken=[key]")]
+    public class Resource1 { /* identical body */ }
+    [TypeForwardedFrom("Foo, Version=1.0.0.0, Culture=neutral, PublicKeyToken=[key]")]
+    public class ResourceData1 { /* identical body */ }
+    [TypeForwardedFrom("Foo, Version=1.0.0.0, Culture=neutral, PublicKeyToken=[key]")]
+    public class Resource1Collection { /* identical body */ }
+}
+```
+
+The PublicKeyToken should match the public key token of the original assembly.
+
+## Extension Methods
+
+In management plane most resources have extension methods on ResourceGroupResource,
+SubscriptionResource, or other common types in Azure.ResourceManager.  When a
+resource is moved to a new package we must also move the extension methods to the
+new package.  To maintain backwards compatibility we will need to change the
+extension methods in the original package by removing the `this` keyword and making
+it a static method that forwards to the new package.  We will also add EBN
+attributes to hide the forwarding methods from IntelliSense.
```

Comment:
this also means the namespace of the FooExtensions and BarExtensions needs to be the same. Otherwise the new extension method might not be picked up in an existing source file.